mobgap/sl/base.py

```diff
@@ -0,0 +1,114 @@
+    sampling_rate_hz
+        The sampling rate of the IMU data in Hz passed to the ``calculate`` method.
+    """,
+        "sl_sec_list_": """
+    sl_sec_list_
+        The main output of the stride length calculation.
+        It provides a DataFrame with the column ``stride_length_m`` that contains the stride length values with one value per full
+        second of the provided data. The unit is ``m``.
+        The index of this dataframe is named ``sec_center_samples`` and contains the sample number of the center of the
+        each second.    """,
+        "stpl_list_": """
+    stpl_list_
+        Secondary output.
+        It provides a Numpy array that contains the raw step length values. The unit is ``m``.
+        """,
+        "calculate_short": """
+    Calculate per-sec stride length values in the passed data.
+    """,
+        "calculate_para": """
+    data
+        The raw IMU data of a single sensor.
+    initial_contacts
+        The indices of the detected initial contacts in the input data.
+    sampling_rate_hz
+        The sampling rate of the IMU data in Hz.
+        """,
+        "calculate_return": """
+    Returns
+    -------
+    self
+        The instance of the class with the ``sl_sec_list_`` attribute set to the estimated stride length per second values.
+    """,
+    },
+    doc_summary="Decorator to fill common parts of the docstring for subclasses of :class:`BaseSlCalculator`.",
+)
+
+
+@base_sl_docfiller
+class BaseSlCalculator(Algorithm):
+    """Base class for SL-calculators.
+
+    This base class should be used for all stride length estimation algorithms.
+    Algorithms should implement the ``calculate`` method, which will perform all relevant processing steps.
+    The method should then return the instance of the class, `sl_sec_list_`` attribute set to the estimated stride length
+    per second values
+    Further, the calculate method should set ``self.data`` and ``self.sampling_rate_hz`` to the parameters passed to the
+    method.
+    We allow that subclasses specify further parameters for the calculate methods (hence, this baseclass supports
+    ``**kwargs``).
+    However, you should only use them, if you really need them and apply active checks, that they are passed correctly.
+    In 99%% of the time, you should add a new parameter to the algorithm itself, instead of adding a new parameter to
+    the calculate method.
+
+    Other Parameters
+    ----------------
+    %(other_parameters)s
+
+    Attributes
+    ----------
+    %(sl_sec_list_)s
+
+    Notes
+    -----
+    You can use the :func:`~base_sl_docfiller` decorator to fill common parts of the docstring for your subclass.
+    See the source of this class for an example.
+    """
+
+    _action_methods = ("calculate",)
+
+    # Other Parameters
+    data: pd.DataFrame
+    initial_contacts: pd.DataFrame
+    sampling_rate_hz: float
+
+    # results
+    sl_sec_list_: pd.DataFrame
+    stpl_list_: np.ndarray
```

Comment:
For the base class I would not include stpl_list_, as we don't assume that all potential stride lnegth algorithms can provide it.

Reply:
You are right, I removed it from the base class!